go/test/endtoend/vreplication/materialize_test.go

```diff
@@ -235,3 +236,103 @@ func TestMaterializeVtctldClient(t *testing.T) {
 		testShardedMaterialize(t, true)
 	})
 }
+
+const (
+	refSchema = `
+  	create table ref1 (
+		id bigint not null,
+		val varbinary(10) not null,
+		primary key (id)
+	) engine=InnoDB default charset=utf8mb4 collate=utf8mb4_unicode_ci;
+	create table ref2 (
+		id bigint not null,
+		id2 bigint not null,
+		primary key (id)
+	) engine=InnoDB default charset=utf8mb4 collate=utf8mb4_unicode_ci;
+`
+	refSourceVSchema = `
+{
+  "tables": {
+    "ref1": {
+      "type": "reference"
+    },
+	"ref2": {
+      "type": "reference"
+	}
+  }
+}
+`
+	refTargetVSchema = `
+{
+  "tables": {
+	"ref1": {
+		  "type": "reference",
+		  "source": "ks1.ref1"
+	},
+	"ref2": {
+		  "type": "reference",
+		  "source": "ks1.ref2"
+	}
+  }
+}
+`
+	initRef1DataQuery = `insert into ks1.ref1(id, val) values (1, 'abc'), (2, 'def'), (3, 'ghi')`
+	initRef2DataQuery = `insert into ks1.ref2(id, id2) values (1, 1), (2, 2), (3, 3)`
+)
+
+// TestReferenceTableMaterialize tests materializing reference tables.
+func TestReferenceTableMaterialize(t *testing.T) {
+	vc = NewVitessCluster(t, nil)
+	require.NotNil(t, vc)
+	defaultReplicas = 0 // because of CI resource constraints we can only run this test with primary tablets
+	defer func() { defaultReplicas = 1 }()
```

Comment:
Better to capture the current value and then reset it to the original in order to prevent unexpected behaviors in the future.